routing/pathfind.go

```diff
@@ -405,8 +406,10 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 			fromVertex, toNode, amountToSend,
 		)
 
-		log.Tracef("path finding probability: fromnode=%v, tonode=%v, "+
-			"probability=%v", fromVertex, toNode, edgeProbability)
+		if log.Level() <= btclog.LevelTrace {
+			log.Tracef("path finding probability: fromnode=%v, tonode=%v, "+
```

Comment:
I haven't got an exact answer. The working theory is that it generates some allocations to hold the var args that are too small to care most of the time but that it adds up because it's called a lot of times per `findPath`

Reply:
That would be easy to test by adding a test function locally in `pathfind.go` with var args and call that instead of `log.Tracef`. You could then also modify it to take four (non var) args and see the difference.

Reply:
It's definitely some overhead due to var args. The fixed arguments version had the same performance as the if. However, the var arg version had the same performance as calling `log.Tracef` directly.

Reply:
Ok interesting. Maybe you could use a `logClosure` here to prevent the if statement.

Reply:
The idea of the closure is that it is only called when the log level is high enough and only then the performance hit is incurred.

```
log.Debug(newLogClosure(func() string {
		return fmt.Sprintf("%v %v %v", a, b, c)
	}))
```

Reply:
And perhaps add a comment here explaining for findings around the var args. So that that experiment is captured for future reference.

Reply:
I would also prefer using a `logClosure`, if that works

Reply:
Just tested log closure and it adds about 1MB of extra allocations and a couple ms of extra runtime. If it is ok with you, I'd keep the if with an explicative comment.

Reply:
Discussed with @halseth offline. For consistency, we prefer the log closure. We can re-evaluate the hot path once the cache changes are in, see how much more significant this becomes.